src/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs

```diff
@@ -181,15 +191,6 @@ public override bool TryComplete(Exception error)
                     ChannelUtilities.Complete(parent._completion, error);
                 }
 
-                // At this point, _blockedWriters and _waitingReaders/Writers will not be mutated:
-                // they're only mutated by readers/writers while holding the lock, and only if _doneWriting is null.
-                // We also know that only one thread (this one) will ever get here, as only that thread
-                // will be the one to transition from _doneWriting false to true.  As such, we can
-                // freely manipulate them without any concurrency concerns.
-                ChannelUtilities.FailInteractors<WriterInteractor<T>, VoidResult>(parent._blockedWriters, ChannelUtilities.CreateInvalidCompletionException(error));
-                ChannelUtilities.WakeUpWaiters(ref parent._waitingReaders, result: false, error: error);
-                ChannelUtilities.WakeUpWaiters(ref parent._waitingWriters, result: false, error: error);
```

Comment:
This change is also potentially problematic, for what the comment below refers to:
"and we do it before completing blocked/waiting readers, so that when they wake up they'll see the task as being completed."
With this change, that's no longer true.